cpp/src/arrow/util/config.h.cmake

```diff
@@ -45,6 +45,7 @@
 #cmakedefine ARROW_IPC
 #cmakedefine ARROW_JSON
 
+#cmakedefine ARROW_GCS
```

Comment:
If we don't have this, `ARROW_GCS` macro isn't defined even when we build Apache Arrow with `-DARROW_GCS=ON`.

If `ARROW_GCS` macro isn't defined, we always get a "Got GCS URI but Arrow compiled without GCS support" error even when we build Apache Arrow with `-DARROW_GCS=ON`.

Reply:
Hmm, I see, does this mean there are no tests for this?